cmake/Translation.cmake

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+set(LY_I18N_BUILD OFF CACHE BOOL "I18N generators")
+set(LY_I18N_LANGUAGE "en_US" CACHE STRING "I18N target language")
+
+# Not all platforms support I18N generators
+if(LY_I18N_BUILD AND NOT PAL_TRAIT_BUILD_HOST_TOOLS)
+    message(ERROR "LY_I18N_BUILD is specified, but not supported for the current target platform")
+    return()
+endif()
+
+# Not all platforms support I18N generators
+if(LY_I18N_BUILD AND NOT LY_I18N_LANGUAGE)
+    message(ERROR "LY_I18N_BUILD is specified, but lack of target language specified.")
+    return()
+endif()
+
+function(add_translation_module target_source_dir module_name)
+    if(NOT LY_I18N_BUILD)
+        return()
+    endif()
+
+    if (PAL_TRAIT_BUILD_HOST_TOOLS)
+        find_package(Qt5 COMPONENTS LinguistTools REQUIRED)
+        find_program(LUPDATE_EXECUTABLE lupdate PATHS "${Qt5_DIR}/../../../bin" REQUIRED)
```

Comment:
The `QT_LRELEASE_EXECUTABLE` cmake variable can be used here directly.

Reply:
`lupdate` is needed here to generate TS files, not `lrelease`.

Reply:
Ok, in that case then the QT_PATH path Cache variable that is set in the [FindQt5.cmake](https://github.com/o3de/3p-package-source/blob/main/package-system/Qt/FindQt.cmake#L15-L19) can be used to get the `lupdate` executable then